### ISSUE TITLE
Room + KSP call converters multiple times, which leads to data corruption

### ISSUE THREAD

**Issue by mreichelt** [google/ksp#779]
When upgrading a project from kapt to ksp I noticed our room unit tests failing. The old tests failed because they tried to read strings like `SomeEnumName` into an `Instant`, which of course would fail. What scratched my head was that the type didn't even mention the `SomeEnumName`.

I was able to create a minimal verifiable example with Room 2.4.0 and KSP 1.5.31-1.0.1 here:
https://github.com/mreichelt/RoomKspBug

To reproduce the bug:

1. Note that the `Awesomeness` enum is missing from [User.kt](https://github.com/mreichelt/RoomKspBug/blob/main/app/src/main/java/com/example/roomkspbug/User.kt)
2. Note that [AppDatabase.kt](https://github.com/mreichelt/RoomKspBug/blob/main/app/src/main/java/com/example/roomkspbug/AppDatabase.kt#L9-L12) has two converters - one for `Instant`, and one for `Awesomeness`
3. Run [InsertUserIntoDatabaseTest](https://github.com/mreichelt/RoomKspBug/blob/main/app/src/test/java/com/example/roomkspbug/InsertUserIntoDatabaseTest.kt) and see that it fails with:
    - `java.time.format.DateTimeParseException: Text 'AWESOME' could not be parsed at index 0`
    - This means that the string `AWESOME` has been written to the `born_at` field! ⚡️🤯
4. The reason this happens is that the generated code in `UserDao_Impl.java` (see locally generated file) contains these lines in the generated `bind` method:

```java
        final String _tmp = __timeConverter.instantToValue(value.getBornAt());      // this is expected
        final Awesomeness _tmp_1 = __awesomenessConverter.valueToAwesomeness(_tmp); // ⚡️ fail - converting the String to Awesomeness
        final String _tmp_2 = __awesomenessConverter.awesomenessToValue(_tmp_1);
```

So basically, the conversion goes like this: `Instant` -> `String` -> `Awesomeness` -> `String`

When run with kapt (see [kapt branch](https://github.com/mreichelt/RoomKspBug/tree/kapt)) the test succeeds, and this is what is written to the `bind` method (as expected):
```java
        final String _tmp = __timeConverter.instantToValue(value.getBornAt());
```

So it seems using KSP leads to data corruption with Room, so this blocks us from migrating to KSP.

**Comment by mreichelt** [comment]
It seems the bug itself is a bug in room, reported here:
https://issuetracker.google.com/issues/206961709

**Comment by yigit** [comment]
I'll close it here since this is a Room issue, not a KSP issue. I'm working on the fix and making your TypeConverters all non-null would fix the issue (that wouldn't work well in kapt but in KSP we can handle that properly).
